database/AddArbeitszeit.java

```diff
@@ -0,0 +1,192 @@
+
+  public boolean addArbeitszeit(String beginTime, String endTime, String totalBreak, String overtime, String comment) throws BatchUpdateException{
+    String query = "INSERT INTO zeitkonto VALUES ('" // Neuer Eintrag wird angelegt
+        + workDate + "', "
+        + MA_ID + ", '"
+        + beginTime + "', '"
+        + endTime + "', '"
+        + totalBreak + "', '"
+        + overtime + "', '"
+        + comment + "')";
+
+        Statement stmt = null;
+        try{
+          connection.setAutoCommit(false); // Das Statement
+          stmt = connection.createStatement();
+          stmt.addBatch(query);
+          stmt.executeBatch();
+          connection.commit();
+          stmt.close();
+          System.out.println("Eintrag erfolgreich angelegt");
+          return true;
+        } catch (BatchUpdateException e){
+          throw new BatchUpdateException();
+        } catch (SQLException sqle){
+          sqle.printStackTrace();
+        } finally {
+          try{
+            if (stmt != null)
+              stmt.close();
+          } catch (SQLException e){
+          }
+        }
+        return false;
+  }
+
+  /**
+   * Ändert an einem Tag für den/die angegebe:n Mitarbeiter:in die Zeiten
+   * Achtung: Wenn der Eintrag nicht existiert, passiert nichts, nicht mal ein Fehler!
+   * Ein SQL-Befehl wird erstellt und an die Datenbank übergeben, wo dieser ausgeführt wird.
+   *
+   * @param beginTime ist der Beginn der Arbeitszeit im Format hh:mm:ss
+   * @param endTime ist das Ende der Arbeitszeit im Format hh:mm:ss
+   * @param totalBreak ist die Gesamtzeit der Pausen an dem Tag im Format hh:mm:ss
+   * @param overtime ist die Anzahl der Überstunden, die an dem Tag geleistet wurden im Format hh:mm:ss
+   * @param comment ist der Kommentar zum Arbeitszeiteintrag
+   */
+
+  public boolean modifyArbeitszeit(String beginTime, String endTime, String totalBreak, String overtime, String comment){
+    String query = "UPDATE zeitkonto " // Eintrag bzw. Einträge werden überarbeitet
+        + "SET Arbeitszeit_Beginn = '" + beginTime + "', "
+        + "Arbeitszeit_Ende = '" + endTime + "', "
+        + "Pausengesamtzeit_Tag = '" + totalBreak + "', "
+        + "Ueberstunden_Tag = '" + overtime + "', "
+        + "Kommentar = '" + comment + "' "
+        + "WHERE work_date = '" + workDate + "' AND " // unter angegebenen Bedingungen
+        + "MA_ID = " + MA_ID;
+
+      Statement stmt = null;
+      try{
+        connection.setAutoCommit(false); // Das Statement
+        stmt = connection.createStatement();
+        stmt.addBatch(query);
+        stmt.executeBatch();
+        connection.commit();
+        stmt.close();
+        System.out.println("Eintrag erfolgreich geändert");
+        return true;
+      } catch (SQLException sqle){
+        sqle.printStackTrace();
+      } finally {
+        try{
+          if (stmt != null)
+            stmt.close();
+        } catch (SQLException e){
+        }
+      }
+      return false;
+  }
```

Comment:
ist das nicht exakt das gleiche wie bei 106-128 --> könnte man das in eine kleine extra Methode stecken, auf die immer wieder zugegriffen werden kann?

Reply:
Ich weiß, das gilt als resolved, aber kurz erklärt: in addWorkingTime wird in dem ähnlichen Block ein Fehler geworfen, den modifyWorkingTime nicht wirft. Ursprünglich hatte ich es aber tatsächlich in einer kleinen extra Methode